BOSS/emscripten/config.txt
{
    "Experiments" : 
    {        
        "Visualize Build Orders" :   
        {   
            "run" : true, 
            "type" : "BuildOrderVisualization", 
            "fps" : 1000, 
            "outputFile" : "CAB_Openers.txt", 
            "scenarios"     :   
            [ 
                { "state":"Protoss Start State", "buildOrder" : "CAB DT Rush" },
                { "state":"Zerg Start State", "buildOrder" : "3 Hatch Muta" }
            ] 
        }
    },

    "States" : 
    {
        "Protoss Start State"   : { "race" : "Protoss", "minerals" : 50, "gas" : 0, "units" : [ ["Protoss_Probe", 4], ["Protoss_Nexus", 1] ] },
        "Zerg Start State"      : { "race" : "Zerg",    "minerals" : 50, "gas" : 0, "units" : [ ["Zerg_Drone", 4], ["Zerg_Hatchery", 1], ["Zerg_Overlord", 1] ] },
        "Terran Start State"    : { "race" : "Terran",  "minerals" : 50, "gas" : 0, "units" : [ ["Terran_SCV", 4], ["Terran_Command_Center", 1] ] }
    },
    
    "Build Orders" :
    {
        "3 Hatch Muta"          : [ "Zerg_Drone", "Zerg_Drone", "Zerg_Drone", "Zerg_Drone", "Zerg_Overlord", "Zerg_Drone", 
                                    "Zerg_Drone", "Zerg_Drone", "Zerg_Drone", "Zerg_Hatchery", "Zerg_Spawning_Pool", "Zerg_Drone", 
                                    "Zerg_Drone", "Zerg_Hatchery", "Zerg_Extractor", "Zerg_Drone", "Zerg_Drone", "Zerg_Drone", "Zerg_Overlord", 
                                    "Zerg_Lair", "Zerg_Extractor", "Zerg_Drone", "Zerg_Drone", "Zerg_Zergling", "Zerg_Zergling", "Zerg_Zergling", 
                                    "Zerg_Drone", "Zerg_Drone", "Zerg_Drone", "Zerg_Drone", "Zerg_Overlord", "Zerg_Drone", "Zerg_Overlord", "Zerg_Drone", 
                                    "Zerg_Overlord", "Zerg_Spire", "Zerg_Drone", "Zerg_Drone", "Zerg_Drone", "Zerg_Drone", "Zerg_Drone", "Zerg_Mutalisk", 
                                    "Zerg_Mutalisk", "Zerg_Mutalisk", "Zerg_Mutalisk", "Zerg_Mutalisk", "Zerg_Mutalisk", "Zerg_Mutalisk", "Zerg_Mutalisk", 
                                    "Zerg_Mutalisk", "Zerg_Mutalisk", "Zerg_Mutalisk", "Zerg_Mutalisk" ],

        "CAB Zealot Rush"       : [ "Protoss_Probe", "Protoss_Probe", "Protoss_Probe", "Protoss_Probe", "Protoss_Pylon", "Protoss_Probe", "Protoss_Gateway", 
                                    "Protoss_Gateway", "Protoss_Probe", "Protoss_Probe", "Protoss_Zealot", "Protoss_Pylon", "Protoss_Zealot", "Protoss_Zealot", 
                                    "Protoss_Probe", "Protoss_Zealot", "Protoss_Zealot", "Protoss_Probe", "Protoss_Pylon", "Protoss_Zealot", "Protoss_Gateway", 
                                    "Protoss_Probe", "Protoss_Pylon", "Protoss_Probe", "Protoss_Zealot", "Protoss_Probe", "Protoss_Zealot", "Protoss_Zealot", 
                                    "Protoss_Zealot", "Protoss_Zealot", "Protoss_Pylon", "Protoss_Probe", "Protoss_Zealot", "Protoss_Zealot", "Protoss_Zealot" ],
        "CAB DT Rush"           : [ "Protoss_Probe", "Protoss_Probe", "Protoss_Probe", "Protoss_Probe", "Protoss_Pylon", "Protoss_Probe", "Protoss_Gateway", 
                                    "Protoss_Probe", "Protoss_Assimilator", "Protoss_Probe", "Protoss_Cybernetics_Core", "Protoss_Probe", "Protoss_Citadel_of_Adun", 
                                    "Protoss_Probe", "Protoss_Templar_Archives", "Protoss_Gateway", "Protoss_Dark_Templar", "Protoss_Dark_Templar", "Protoss_Pylon", 
                                    "Protoss_Dark_Templar", "Protoss_Dark_Templar", "Protoss_Probe", "Protoss_Pylon", "Protoss_Probe"],
        "CAB Dragoon Rush"      : [ "Protoss_Probe", "Protoss_Probe", "Protoss_Probe", "Protoss_Probe", "Protoss_Pylon", "Protoss_Probe", "Protoss_Probe", 
                                    "Protoss_Gateway", "Protoss_Probe", "Protoss_Assimilator", "Protoss_Probe", "Protoss_Probe", "Protoss_Cybernetics_Core", 
                                    "Protoss_Probe", "Protoss_Probe", "Protoss_Gateway", "Singularity_Charge", "Protoss_Dragoon", "Protoss_Gateway", "Protoss_Pylon", 
                                    "Protoss_Dragoon", "Protoss_Dragoon", "Protoss_Probe", "Protoss_Gateway", "Protoss_Pylon", "Protoss_Probe", "Protoss_Dragoon", 
                                    "Protoss_Dragoon", "Protoss_Dragoon"]
    }
}

CasiaBot/CasiaBot_Config.txt
{
    "Bot Info" :
    {
        "BotName"                   : "CasiaBot",
        "Authors"                   : "David Churchill",
        "PrintInfoOnStart"          : false
    },
        
    "BWAPI" : 
    {
        "SetLocalSpeed"             : 3,
        "SetFrameSkip"              : 0,
        "UserInput"                 : true,
        "CompleteMapInformation"    : false
    },
    
    "Micro" :
    {
        "UseSparcraftSimulation"    : true,
        "KiteWithRangedUnits"       : true,
        "KiteLongerRangedUnits"     : ["Mutalisk", "Vulture"],
        "WorkersDefendRush"         : true,
        "RetreatMeleeUnitShields"   : 0,
        "RetreatMeleeUnitHP"        : 0,
        "InCombatRadius"            : 1000,
        "RegroupRadius"             : 300,
        "UnitNearEnemyRadius"       : 600
    },
    
    "Macro" :
    {
        "BOSSFrameLimit"            : 160,
        "WorkersPerRefinery"        : 3,
        "BuildingSpacing"           : 1,
        "PylonSpacing"              : 3
    },

    "Debug" :
    {
        "ErrorLogFilename"          : "bwapi-data/AI/CasiaBot_ErrorLog.txt",
        "LogAssertToErrorFile"      : false,
        
        "DrawGameInfo"              : false,   
        "DrawUnitHealthBars"        : true,
        "DrawProductionInfo"        : true, 
        "DrawBuildOrderSearchInfo"  : false,
        "DrawScoutInfo"             : false,
        "DrawEnemyUnitInfo"         : false,
        "DrawModuleTimers"          : false,
        "DrawResourceInfo"          : true,
        "DrawCombatSimInfo"         : false,
        "DrawUnitTargetInfo"        : false,
        "DrawBWTAInfo"              : false,
        "DrawMapGrid"               : false,
        "DrawSquadInfo"             : false,
        "DrawSquadTypeInfo"         : true,
        "DrawWorkerInfo"            : true,
        "DrawMouseCursorInfo"       : false,
        "DrawBuildingInfo"          : false,
        "DrawReservedBuildingTiles" : false,
        "DrawBOSSStateInfo"         : false,
        "PrintModuleTimeout"        : false
    },
    
    "Modules" :
    {
        "UseGameCommander"          : true,
        "UseScoutManager"           : true,
        "UseCombatCommander"        : true,
        "UseBuildOrderSearch"       : true,
        "UseOpeningIO"              : true,
        "UseAutoObserver"           : false,
        "UsingOpeningIO"            : false
    },
    
    "Tools" :
    {
        "MapGridSize"               : 320
    },
    
    "Opening" :
    {        
        "ZVP"                       : "ZVP_9D",
        "ZVT"                       : "ZVT_9D",
        "ZVZ"                       : "ZVZ_9D",
        "Random"                    : "ZVT_5D",
        
        "ScoutGasSteal"             : false,
        "ScoutHarassEnemy"          : true,
        
        "ReadDirectory"             : "bwapi-data/read/",
        "WriteDirectory"            : "bwapi-data/write/",
                
        "UseEnemySpecificOpening"  : true,
        "EnemySpecificOpening"     :
        {
            "LetaBot"               : "ZVP_5D"
        },
        
        "Openings" :
        {
            "ZVP_5D"        : { "OpeningBuildOrder" : ["Drone", "Spawning_Pool", "Drone", "Drone", "Zergling", "Zergling", "Zergling"] },

            "ZVP_9D_Speed"  : { "OpeningBuildOrder" : ["Drone", "Drone", "Drone", "Drone", "Drone", "Spawning_Pool", "Drone", "Extractor", "Overlord", "Drone", "Zergling", "Zergling", "Zergling", "Metabolic_Boost", "Zergling"] },

            "ZVP_9D"  : { "OpeningBuildOrder" : ["Drone", "Drone", "Drone", "Drone", "Drone", "Spawning_Pool", "Drone", "Extractor", "Drone", "Extractor@Cancel", "Overlord", "Zergling", "Zergling", "Zergling", "Hatchery", "Zergling"] },

            "ZVP_Overpool"  : { "OpeningBuildOrder" : ["Drone", "Drone", "Drone", "Drone", "Drone", "Overlord", "Spawning_Pool", "Drone", "Drone", "Drone", "Zergling", "Zergling", "Zergling", "Hatchery", "Hatchery"] },

            "ZVP_10Hatch"  : { "OpeningBuildOrder" : ["Drone", "Drone", "Drone", "Drone", "Drone", "Extractor", "Drone", "Extractor@Cancel", "Hatchery", "Spawning_Pool", "Drone", "Overlord", "Extractor", "Drone", "Extractor@Cancel", "Hatchery", "Zergling", "Zergling", "Zergling", "Zergling", "Overlord"] },

            "ZVP_3Hatch"  : { "OpeningBuildOrder" : ["Drone", "Drone", "Drone", "Drone", "Drone", "Overlord", "Drone", "Drone", "Drone", "Hatchery", "Spawning_Pool", "Drone", "Drone", "Drone", "Drone", "Hatchery", "Zergling", "Zergling", "Zergling", "Zergling", "Extractor", "Overlord", "Hydralisk_Den", "Muscular_Augments", "Extractor", "Hydralisk", "Hydralisk", "Hydralisk", "Hydralisk", "Grooved_Spines"] },

            "ZVP_3Hatch_Before_Pool"  : { "OpeningBuildOrder" : ["Drone", "Drone", "Drone", "Drone", "Drone", "Overlord", "Drone", "Drone", "Drone", "Hatchery", "Hatchery", "Spawning_Pool"] },

            "ZVP_4Hatch_Before_Gas"  : { "OpeningBuildOrder" : ["Drone", "Drone", "Drone", "Drone", "Drone", "Overlord", "Drone", "Drone", "Drone", "Hatchery", "Spawning_Pool", "Drone", "Drone", "Hatchery", "Zergling", "Zergling", "Zergling", "Zergling", "Drone", "Hatchery", "Extractor", "Drone", "Drone", "Evolution_Chamber", "Lair", "Metabolic_Boost", "Zerg_Carapace"] },

            "ZVP_1_Hatch_Lurker"  : { "OpeningBuildOrder" : ["Drone", "Drone", "Drone", "Drone", "Drone", "Overlord", "Drone", "Drone", "Extractor", "Spawning_Pool", "Drone", "Zergling", "Zergling", "Zergling", "Metabolic_Boost", "Lair", "Drone", "Drone", "Drone", "Drone", "Overlord", "Hydralisk_Den", "Drone", "Zergling", "Drone", "Zergling", "Hydralisk", "Hydralisk", "Hydralisk", "Hydralisk", "Lurker_Aspect", "Overlord", "Lurker", "Lurker", "Lurker", "Lurker"] },

            "ZVP_2_Hatch_Lurker"  : { "OpeningBuildOrder" : ["Drone", "Drone", "Drone", "Drone", "Drone", "Extractor", "Drone", "Extractor@Cancel", "Overlord", "Drone", "Drone", "Hatchery", "Spawning_Pool", "Drone", "Drone", "Zergling", "Zergling", "Zergling", "Zergling", "Extractor", "Drone", "Overlord", "Drone", "Metabolic_Boost", "Zergling", "Zergling", "Zergling", "Extractor", "Creep_Colony", "Sunken_Colony", "Lair", "Hydralisk_Den", "Drone", "Drone", "Drone", "Drone", "Overlord", "Zergling", "Lurker_Aspect", "Hydralisk", "Hydralisk", "Hydralisk", "Lurker", "Lurker", "Lurker", "Lurker"] },

            "ZVP_2_Hatch_Muta"  : { "OpeningBuildOrder" : ["Drone", "Drone", "Drone", "Drone", "Drone", "Overlord", "Drone", "Drone", "Drone", "Hatchery", "Drone", "Spawning_Pool", "Drone", "Extractor", "Drone", "Zergling", "Zergling", "Zergling", "Zergling", "Lair", "Overlord", "Metabolic_Boost", "Drone", "Drone", "Drone", "Drone", "Drone", "Drone", "Drone", "Drone", "Drone", "Drone", "Spire", "Hatchery", "Extractor", "Overlord", "Overlord", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Overlord", "Hatchery"] },

            "ZVT_5D" : { "OpeningBuildOrder" : ["Drone", "Spawning_Pool", "Drone", "Drone", "Zergling", "Zergling", "Zergling", "Extractor@80%Zergling", "Drone", "Extractor@Cancel", "Overlord", "Zergling", "Zergling", "Zergling", "Zergling", "Zergling", "Zergling", "Zergling", "Zergling", "Hatchery@300M"] },

            "ZVT_12Hatch" : { "OpeningBuildOrder" : ["Drone", "Drone", "Drone", "Drone", "Drone", "Overlord", "Drone", "Drone", "Drone", "Hatchery", "Spawning_Pool"] },

            "ZVT_12D" : { "OpeningBuildOrder" : ["Drone", "Drone", "Drone", "Drone", "Drone", "Overlord", "Drone", "Drone", "Drone", "Hatchery", "Zergling", "Zergling", "Zergling", "Drone", "Drone", "Drone", "Hatchery", "Overlord", "Drone", "Drone", "Creep_Colony", "Sunken_Colony"] },

            "ZVT_9D" : { "OpeningBuildOrder" : ["Drone", "Drone", "Drone", "Drone", "Drone", "Spawning_Pool", "Drone", "Extractor", "Drone", "Extractor@Cancel", "Overlord", "Zergling", "Zergling", "Zergling"] },

            "ZVT_3Hatch_Muta" : { "OpeningBuildOrder" : ["Drone", "Drone", "Drone", "Drone", "Drone", "Overlord", "Drone", "Drone", "Drone", "Hatchery", "Spawning_Pool", "Drone", "Drone", "Drone", "Hatchery@Main", "Extractor", "Zergling", "Zergling", "Drone", "Drone", "Drone", "Overlord", "Lair", "Zergling", "Drone", "Drone", "Metabolic_Boost", "Extractor", "Drone", "Zergling", "Drone", "Zergling", "Spire", "Zergling", "Drone", "Drone", "Overlord", "Drone", "Drone", "Drone", "Zergling", "Zergling", "Zergling", "Zergling", "Zergling", "Zergling", "Creep_Colony", "Creep_Colony", "Sunken_Colony", "Sunken_Colony", "Overlord@50%Spire", "Overlord", "Overlord", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Hydralisk_Den", "Mutalisk", "Mutalisk", "Overlord", "Evolution_Chamber@75%Hydralisk_Den", "Lurker_Aspect"] },

            "ZVT_2Hatch_Muta" : { "OpeningBuildOrder" : ["Drone", "Drone", "Drone", "Drone", "Drone", "Overlord", "Drone", "Drone", "Drone", "Hatchery", "Spawning_Pool", "Extractor", "Zergling", "Zergling", "Zergling", "Lair", "Metabolic_Boost", "Drone", "Drone", "Drone", "Drone", "Spire", "Overlord", "Overlord", "Extractor", "Drone", "Drone", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk"] },

            "ZVT_3Hatch_Lurker" : { "OpeningBuildOrder" : ["Drone", "Drone", "Drone", "Drone", "Drone", "Overlord", "Drone", "Drone", "Drone", "Hatchery", "Spawning_Pool", "Drone", "Drone", "Drone", "Hatchery@Main", "Zergling", "Zergling", "Zergling", "Drone", "Overlord", "Extractor", "Lair", "Drone", "Drone", "Drone", "Zergling", "Drone", "Drone", "Drone", "Overlord", "Metabolic_Boost", "Hydralisk_Den@50%Lair", "Extractor@90%Lair", "Lurker_Aspect", "Evolution_Chamber", "Hydralisk@50%Lurker_Aspect", "Hydralisk", "Hydralisk", "Hydralisk", "Hydralisk", "Hydralisk", "Hydralisk", "Zerg_Carapace", "Lurker", "Lurker", "Lurker", "Lurker", "Lurker", "Lurker", "Lurker"] },

            "ZVT_2Hatch_Lurker" : { "OpeningBuildOrder" : ["Drone", "Drone", "Drone", "Drone", "Drone", "Overlord", "Drone", "Drone", "Drone", "Hatchery", "Spawning_Pool", "Extractor", "Drone", "Drone", "Drone", "Zergling", "Zergling", "Zergling", "Zergling", "Overlord", "Extractor", "Lair", "Drone", "Drone", "Drone", "Zergling", "Drone", "Drone", "Drone", "Overlord", "Metabolic_Boost", "Hydralisk_Den@50%Lair", "Extractor@90%Lair", "Lurker_Aspect", "Evolution_Chamber", "Hydralisk@50%Lurker_Aspect", "Hydralisk", "Hydralisk", "Hydralisk", "Hydralisk", "Hydralisk", "Hydralisk", "Zerg_Carapace", "Lurker", "Lurker", "Lurker", "Lurker", "Lurker"] },

            "ZVZ_5D" : { "OpeningBuildOrder" : ["Drone", "Spawning_Pool", "Drone", "Drone", "Zergling", "Zergling", "Zergling"] },

            "ZVZ_9D" : { "OpeningBuildOrder" : ["Drone", "Drone", "Drone", "Drone", "Drone", "Spawning_Pool", "Drone", "Extractor", "Overlord", "Drone", "Zergling", "Zergling", "Zergling"] },

            "ZVZ_Overpool" : { "OpeningBuildOrder" : ["Drone", "Drone", "Drone", "Drone", "Drone", "Overlord", "Spawning_Pool", "Drone", "Extractor", "Drone", "Drone", "Zergling", "Zergling", "Zergling"] },

            "ZVZ_12D"     : { "OpeningBuildOrder" : ["Drone", "Drone", "Drone", "Drone", "Drone", "Overlord", "Drone", "Drone", "Drone", "Spawning_Pool", "Drone", "Extractor", "Hatchery", "Zergling", "Zergling", "Zergling", "Metabolic_Boost", "Lair", "Spire"] },

            "ZVZ_12Hatch" : { "OpeningBuildOrder" : ["Drone", "Drone", "Drone", "Drone", "Drone", "Overlord", "Drone", "Drone", "Drone", "Hatchery", "Spawning_Pool", "Extractor", "Drone", "Drone", "Drone", "Zergling", "Zergling", "Zergling", "Lair", "Zergling", "Overlord", "Metabolic_Boost"] },

            "ZVZ_11Gas_10Pool" : { "OpeningBuildOrder" : ["Drone", "Drone", "Drone", "Drone", "Drone", "Overlord", "Drone", "Drone", "Extractor", "Spawning_Pool", "Drone", "Drone", "Zergling", "Zergling", "Zergling", "Lair", "Metabolic_Boost", "Zergling", "Zergling", "Overlord", "Spire"] },

            "ZVZ_9D_Speed_1Hatch_Spire" : { "OpeningBuildOrder" : ["Drone", "Drone", "Drone", "Drone", "Drone", "Spawning_Pool", "Drone", "Extractor", "Overlord", "Drone", "Zergling", "Zergling", "Zergling", "Metabolic_Boost", "Lair", "Zergling", "Zergling", "Zergling", "Zergling", "Overlord", "Spire"] }
        }
    }
}

CasiaBot/Source/Config.cpp
#include "Config.h"
#include "CABAssert.h"

namespace Config
{
    namespace ConfigFile
    {
        bool ConfigFileFound                = false;
        bool ConfigFileParsed               = false;
		std::string ConfigFileLocation = "bwapi-data/AI/CasiaBot_Config.txt";
    }

    namespace Opening
    {
        std::string OpeningName            = "ZVP_9D";
        std::string ReadDir                 = "bwapi-data/read/";
        std::string WriteDir                = "bwapi-data/write/";
        bool GasStealWithScout              = false;
        bool ScoutHarassEnemy               = true;
        bool UseEnemySpecificOpening       = false;
        bool FoundEnemySpecificOpening     = false;
    }

    namespace Modules							    
    {
        // the default tournament bot modules
        bool UsingGameCommander             = true;     // toggle GameCommander, effectively CasiaBot
        bool UsingScoutManager              = true;
        bool UsingCombatCommander           = true;
        bool UsingBuildOrderSearch          = true;     // toggle use of Build Order Search, currently no backup
        bool UsingAutoObserver              = false;
        bool UsingOpeningIO					= true;    // toggle the use of file io for strategy
        bool UsingUnitCommandManager        = false;    // handles all unit commands
		
        // extra things, don't enable unless you know what they are
        bool UsingBuildOrderDemo            = false;
    }

    namespace BotInfo
    {
        std::string BotName                 = "CasiaBot";
        std::string Authors                 = "Dave Churchill";
        bool PrintInfoOnStart               = false;
    }

    namespace BWAPIOptions
    {
        int SetLocalSpeed                   = 42;
        int SetFrameSkip                    = 0;
        bool EnableUserInput                = true;
        bool EnableCompleteMapInformation   = false;
    }
    
    namespace Tournament						
    {
        int GameEndFrame                    = 86400;	
    }
    
    namespace Debug								
    {
        bool DrawGameInfo                   = true;
        bool DrawUnitHealthBars             = true;
        bool DrawProductionInfo             = true;
        bool DrawBuildOrderSearchInfo       = false;
        bool DrawScoutInfo                  = false;
        bool DrawResourceInfo               = true;
        bool DrawWorkerInfo                 = true;
        bool DrawModuleTimers               = false;
        bool DrawReservedBuildingTiles      = false;
        bool DrawCombatSimulationInfo       = false;
        bool DrawBuildingInfo               = false;
        bool DrawMouseCursorInfo            = false;
        bool DrawEnemyUnitInfo              = false;
        bool DrawBWTAInfo                   = false;
        bool DrawMapGrid                    = false;
        bool DrawUnitTargetInfo             = false;
        bool DrawSquadInfo                  = false;
		bool DrawSquadTypeInfo				= true;
        bool PrintModuleTimeout             = false;	

        std::string ErrorLogFilename        = "CAB_ErrorLog.txt";
        bool LogAssertToErrorFile           = false;

        BWAPI::Color ColorLineTarget        = BWAPI::Colors::White;
        BWAPI::Color ColorLineMineral       = BWAPI::Colors::Cyan;
        BWAPI::Color ColorUnitNearEnemy     = BWAPI::Colors::Red;
        BWAPI::Color ColorUnitNotNearEnemy  = BWAPI::Colors::Green;
    }

    namespace Micro								
    {
        bool UseSparcraftSimulation         = true;
        bool KiteWithRangedUnits            = true;
        std::set<BWAPI::UnitType> KiteLongerRangedUnits;
        bool WorkersDefendRush              = false; 
		int RetreatMeleeUnitShields         = 0;
        int RetreatMeleeUnitHP              = 0;
        int CombatRadius                    = 1000;     // radius of combat to consider units for Micro Search
        int CombatRegroupRadius             = 300;      // radius of units around frontmost unit we consider in regroup calculation
        int UnitNearEnemyRadius             = 600;      // radius to consider a unit 'near' to an enemy unit
    }

    namespace Macro
    {
        int WorkersPerRefinery              = 3;
        int WorkersPerMineralPatch          = 3;
        int BuildingSpacing                 = 1;
        int PylonSpacing                    = 3;
    }

    namespace Tools								
    {
        extern int MAP_GRID_SIZE            = 320;      // size of grid spacing in MapGrid
    }
}